src/pages/index.js

```diff
@@ -45,40 +89,106 @@ addCardMOB.addEventListener("click", () => {
   addPlacePopup.open();
 });
 
+profilePic.addEventListener("click", () => {
+  profilePicPopup.open();
+});
+
 //***************************************************************************************************************************//
 //                                             Event Handlers                                                                //
 //***************************************************************************************************************************//
 
 function handleEditProfile(userData) {
+  const submitButton = editProfileModal.querySelector(
+    "#edit-profile-modal-submit-button"
+  );
+  submitButton.textContent = "Saving...";
+  api.patchProfileInfo(userData.name, userData.job);
   userInfo.setUserInfo(userData);
```

Comment:
You need to update `DOM` only in blocks `then` after a successful response from the server so that when there is an error it wouldn’t change